src/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -11,19 +11,27 @@ namespace System.Diagnostics
 {
     internal static class TraceInternal
     {
+        static TraceInternal()
+        {
+            // This is where we override default DebugProvider because we know
+            // for sure that we have some Listeners to write to.
+            s_originalProvider = Debug.SetProvider(s_provider);
+        }
+
         private class TraceProvider : DebugProvider
         {
+            public override int IndentLevel { get { return TraceInternal.IndentLevel; } set { TraceInternal.IndentLevel = value; } }
```

Comment:
It should not be necessary to override the getter for these. Both the base class implementation and the overrriden implementation will end up going to `DebugProvider.s_indentLevel` now.

It may be more sensible to have a `virtual void OnIndentLevelChanged(int indentLevel)` that you override here and remove the IndentLevel property getter/setter.

Same for `IndentSize`.

Reply:
Made this change in the new PRs